token/confidential-transfer/elgamal-registry/src/processor.rs

```diff
@@ -0,0 +1,191 @@
+    )?;
+
+    let elgamal_registry_account_data = &mut elgamal_registry_account_info.data.borrow_mut();
+    let elgamal_registry_account =
+        pod_from_bytes_mut::<ElGamalRegistry>(elgamal_registry_account_data)?;
+    elgamal_registry_account.owner = *wallet_account_info.key;
+    elgamal_registry_account.elgamal_pubkey = proof_context.pubkey;
+
+    Ok(())
+}
+
+/// Processes `UpdateRegistry` instruction
+pub fn process_update_registry_account(
+    _program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    proof_instruction_offset: i64,
+) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+    let elgamal_registry_account_info = next_account_info(account_info_iter)?;
+    let elgamal_registry_account_data = &mut elgamal_registry_account_info.data.borrow_mut();
+    let elgamal_registry_account =
+        pod_from_bytes_mut::<ElGamalRegistry>(elgamal_registry_account_data)?;
+
+    // zero-knowledge proof certifies that the supplied ElGamal public key is valid
+    let proof_context = verify_and_extract_context::<
+        PubkeyValidityProofData,
+        PubkeyValidityProofContext,
+    >(account_info_iter, proof_instruction_offset, None)?;
+
+    let owner_info = next_account_info(account_info_iter)?;
+    validate_owner(owner_info, &elgamal_registry_account.owner)?;
+
+    elgamal_registry_account.elgamal_pubkey = proof_context.pubkey;
+    Ok(())
+}
+
+/// Instruction processor
+pub fn process_instruction(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    input: &[u8],
+) -> ProgramResult {
+    let instruction = RegistryInstruction::unpack(input)?;
+    match instruction {
+        RegistryInstruction::CreateRegistry {
+            proof_instruction_offset,
+        } => {
+            msg!("RegistryInstruction::CreateRegistry");
+            process_create_registry_account(program_id, accounts, proof_instruction_offset as i64)
+        }
+        RegistryInstruction::UpdateRegistry {
+            proof_instruction_offset,
+        } => {
+            msg!("RegistryInstruction::UpdateRegistry");
+            process_update_registry_account(program_id, accounts, proof_instruction_offset as i64)
+        }
+    }
+}
+
+fn validate_owner(owner_info: &AccountInfo, expected_owner: &Pubkey) -> ProgramResult {
+    if expected_owner != owner_info.key {
+        return Err(ProgramError::InvalidAccountOwner);
+    }
+    if !owner_info.is_signer {
+        return Err(ProgramError::MissingRequiredSignature);
+    }
+    Ok(())
+}
+
+/// Creates ElGamal registry account using Program Derived Address for the given
+/// seeds
+pub fn create_pda_account<'a>(
+    payer: &AccountInfo<'a>,
+    rent: &Rent,
+    space: usize,
+    owner: &Pubkey,
+    system_program: &AccountInfo<'a>,
+    new_pda_account: &AccountInfo<'a>,
+    new_pda_signer_seeds: &[&[u8]],
+) -> ProgramResult {
+    if new_pda_account.lamports() > 0 {
+        let required_lamports = rent
+            .minimum_balance(space)
+            .max(1)
```

Comment:
Why is the `max(1)` needed?

Reply:
Oh oops yeah I was going to ask this in the PR description, but I completely forgot. I got this function from the [ATA program](https://github.com/solana-labs/solana-program-library/blob/master/associated-token-account/program/src/tools/account.rs#L31) and it has this max value, which I wasn't sure what it was achieving.

Reply:
Ah, I wonder if that's there while testing with 0 rent. I would go with removing the logic here, since I hope our testing tools are better now and never provide 0 rent.